### PR TITLE
fix(provider/wikipedia): unescape HTML entities in cleanMarkup (#1847)

### PR DESCRIPTION
**Pull request by sydlexius**
Wikipedia API responses embed HTML entities (e.g. `&amp;`, `&lt;`, `&#160;`) in plain-text fields. The `cleanMarkup` function was stripping markup but not unescaping these entities, causing them to appear literally in stored biography text and propagate to origin, genres, members, and past_members fields. This PR adds a single `html.UnescapeString` call in `cleanMarkup` to fix all affected fields at once.

Closes #1847

## Verification

- `html.UnescapeString` applied in `cleanMarkup`; propagates to origin/genres/members/past_members
- 6 new table-driven test cases covering entity variants (`&amp;`, `&lt;`, `&gt;`, `&#160;`, mixed)
- `internal/provider/wikipedia` tests pass; build and lint clean; patch coverage 100% (1/1)
- Pre-push gate GREEN

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of HTML entities and angle-bracket text in Wikipedia infobox content so special characters, spaces, and literal < or > are preserved and decoded correctly.

* **Tests**
  * Expanded tests to cover named and numeric HTML entity decoding, entity edge-cases after links, and preservation of embedded angle-bracket text.

### CODE REVIEW

File: internal/provider/wikipedia/infobox_test.go [hunk 1]
```diff
@@ -203,6 +203,12 @@ func TestCleanMarkup(t *testing.T) {
 		{"self-closing ref", "text<ref name=foo />more", "textmore"},
 		{"html tags", "<b>bold</b> and <i>italic</i>", "bold and italic"},
 		{"plain text", "just plain text", "just plain text"},
+		{"named entity amp", "&amp;", "&"},
+		{"named entity accent", "na&iuml;ve", "naïve"},
+		{"entity nbsp in text", "New&nbsp;York", "New York"},
+		{"numeric entity apostrophe", "rock&#39;n&#39;roll", "rock'n'roll"},
+		{"numeric entity en-dash", "1990&#8211;2000", "1990–2000"},
+		{"entity mixed with wikilink", "[[New York]]&nbsp;City", "New York City"},
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/provider/wikipedia/infobox_test.go:211]: Fixed in 727ac4d5: added test cases asserting decoded angle brackets survive (&lt; -> <, &gt; -> >, and a regression case 'born 1 &lt; 2 years ago').

File: internal/provider/wikipedia/infobox.go [hunk 2]
```diff
@@ -363,6 +363,7 @@ func unwrapListTemplates(s string) string {
 // Handles: [[Link|Display]] -> Display, [[Link]] -> Link,
 // {{nowrap|text}} -> text, {{small|text}} -> text, HTML tags, ref tags.
 func cleanMarkup(s string) string {
+	s = html.UnescapeString(s)
 	s = stripRefs(s)
 	s = stripHTMLTags(s)
 	s = resolveWikilinks(s)
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/provider/wikipedia/infobox.go:369]: Fixed in 727ac4d5: hardened stripHTMLTags to only enter tag-skip mode when '<' is followed by a valid HTML tag start char (letter, /, !, ?); a bare '<' (e.g. before a space/digit) now passes through as a literal. Also fixed the symmetric bare-'>' case. The truncation example 'born 1 < 2 years ago' now round-trips correctly.
